state-chain/pallets/cf-ingress-egress/src/tests.rs

```diff
@@ -1794,3 +1794,34 @@ fn ingress_fee_is_withheld_or_scheduled_for_swap() {
 
 	test_ingress_or_egress_fee_is_withheld_or_scheduled_for_swap(ingress_function)
 }
+
+#[test]
+fn safe_mode_prevents_deposit() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(IngressEgress::open_channel(
+			&ALICE,
+			eth::Asset::Eth,
+			ChannelAction::LiquidityProvision { lp_account: 0 },
```

Comment:
Would be another PR - but perhaps it makes sense to block deposit channels for swaps, but not LPs e.g. in a liquidity shortage or something we might want to do that? Something we can think about